crud/select/iterator.lua

```diff
@@ -1,196 +1,223 @@
-local errors = require('errors')
-local fiber = require('fiber')
-
-local dev_checks = require('crud.common.dev_checks')
-local utils = require('crud.common.utils')
-
-local UpdateTuplesError = errors.new_class('UpdateTuplesError')
-local GetTupleError = errors.new_class('GetTupleError')
-
-local Heap = require('crud.common.heap')
-
-local Iterator = {}
-Iterator.__index = Iterator
-
-function Iterator.new(opts)
-    dev_checks({
-        space_name = 'string',
-        space_format = 'table',
-        comparator = 'function',
-        iteration_func = 'function',
-
-        plan = 'table',
-
-        batch_size = 'number',
-        replicasets = 'table',
-
-        timeout = '?number',
-    })
-
-    local iter = {
-        space_name = opts.space_name,
-        space_format = opts.space_format,
-        iteration_func = opts.iteration_func,
-
-        plan = opts.plan,
-
-        timeout = opts.timeout,
-
-        replicasets = table.copy(opts.replicasets),
-        replicasets_count = utils.table_count(opts.replicasets),
-        empty_replicasets = {},
-        empty_replicasets_count = 0,
-
-        batch_size = opts.batch_size,
-
-        tuples_by_replicasets = {},
-        next_tuple_indexes = {},
+local buffer = require('buffer')
+local msgpack = require('msgpack')
+local log = require('log')
+local ffi = require('ffi')
+local collations = require('crud.common.collations')
+
+local key_def_lib
+local merger_lib
+
+if pcall(require, 'tuple.merger') then
+    merger_lib = require('tuple.merger')
+    key_def_lib = require('tuple.keydef')
+elseif pcall(require, 'merger') then
+    log.info('Impossible to load "tuple-merger" module. Use built-in "merger"')
+    merger_lib = require('merger')
+    key_def_lib = require('key_def')
+else
+    error('Seems your Tarantool version (' .. _TARANTOOL ..
+            ') does not support "tuple-merger" or "merger" modules')
+end
 
-        heap = Heap.new({ comparator = opts.comparator }),
-        tuples_count = 0,
+local SELECT_FUNC_NAME = '_crud.select_on_storage'
```

Comment:
I seems that this function name should be defined in `select` module (where function is defined) and passed to constructor.